systemtest/src/main/java/io/strimzi/systemtest/storage/TestStorage.java

```diff
@@ -35,6 +35,8 @@ final public class TestStorage {
     private static final String ADMIN = "admin-client";
     private static final String USER = "user";
     private static final String CLUSTER_NAME_PREFIX = "my-cluster-";
+    private static final String BROKER_ROLE_PREFIX = "broker-";
+    private static final String CONTROLLER_ROLE_PREFIX = "controller-";
```

Comment:
What is this used for?

Reply:
I see the tests seem to generate node pools with random names (using the hash of the cluster name but adding this prefix). But maybe @im-konge can provide more details about this.

Reply:
that's to our testcases, to distinct between broker and controller NodePools in the tests (and during the debug, so we don't have to look into spec of the particular nodepool and know, which role is used there)

Reply:
Most users will go through the following chain:
1) Convert from Kafka to NodePools
2) Migrate to KRaft

That will leave them with broker pool named `kafka`. It would be good to have this covered in the STs as a single test or as a ST that starts with how the cluster will look like after migration to node pools. (I do not really care how the pools are named in the other tests, but I think we should have this chain of events covered.)

Reply:
So for brokers we can have `kafka` pool, but the controller's NodePool should anyway have different name, right? I can change it in the tests so it follows the use-case you mentioned.

Reply:
Good test case raised by Jakub. If their Kafka cluster aren't using node pools yet, they have to migrate to them first and the name for broker pool will be `kafka`.
Regarding the controllers pool, it can have the name you prefer.

Reply:
@im-konge are you planning to work on such a ST? wdyt?

Reply:
`kafka` nodepool is already in place instead of `broker-xxx` nodepool
so I guess there is not much to work on
or you have different testcase on your mind?

Reply:
AFAIU what @scholzj said is about starting from a ZooKeeper-based cluster without node pools for brokers.
Running the migration to node pools (still ZooKeeper-based cluster) then running the KRaft migration.

Reply:
I can do that, but the movement from ZK without nodepools to ZK with nodepool scenario is tested somewhere else. And if it doesn't have any impact on the migration itself, I guess we would just extend the test execution without any benefit -> because it's already tested and this should really focus only on migration.
But if it makes some difference, I can add check for it